Orange/data/table/base.py

```diff
@@ -0,0 +1,1409 @@
+
+        # transfer attrs from self
+        result.__finalize__(self)
+
+        # fix multiple identical columns
+        for icn in self._INTERNAL_COLUMN_NAMES:
+            # duplicated cols are appended with _x and _y by default
+            matching_columns = [c for c in result.columns if icn in c]
+            first_col = result[matching_columns[0]]
+            result.drop(matching_columns, axis=1, inplace=True)
+            result[icn] = first_col
+
+        # process a list of variables, appending suffix if not found in the
+        # resulting columns (that means it was a dup)
+        def suffix_dups(varlist, suffix):
+            return [v if v in result.columns else v.copy(new_name=v.name + suffix) for v in varlist]
+
+        # dedup because the target join valriable doesn't get renamed and there is
+        # only one column, wehile without this, the domain would have two
+        def dedup_inorder(varlist):
+            s = set()
+            return [x for x in varlist if not (x in s or s.add(x))]
+
+        # merge domain
+        new_domain = Domain(
+            dedup_inorder(suffix_dups(self.domain.attributes, suffix_left) +
+                          suffix_dups(right.domain.attributes, suffix_right)),
+            dedup_inorder(suffix_dups(self.domain.class_vars, suffix_left) +
+                          suffix_dups(right.domain.class_vars, suffix_right)),
+            dedup_inorder(suffix_dups(self.domain.metas, suffix_left) +
+                          suffix_dups(right.domain.metas, suffix_right))
+        )
+        result.domain = new_domain
+
+        # always reindex
+        result.index = self._new_id(len(result), force_list=True)
+
+        return result
+
+    def approx_len(self):
+        """Return the approximate length of the table."""
+        # TODO: remove, here solely for SQL compatibility, which would ideally be replaced with Spark
+        # https://github.com/biolab/orange3/wiki/pandas-migration#future-todos-and-architectural-wishlist
+        return len(self)
+
+    def exact_len(self):
+        """Return the exact length of the table."""
+        # TODO: remove, here solely for SQL compatibility, which would ideally be replaced with Spark
+        # https://github.com/biolab/orange3/wiki/pandas-migration#future-todos-and-architectural-wishlist
+        return len(self)
+
+    def has_missing(self):
+        """Return `True` if there are any missing attribute or class values."""
+        # manual access to columns because dumping to a numpy array (with self.X) is slower
+        return self[self.domain.attributes].isnull().any().any() or self.has_missing_class()
+
+    def has_missing_class(self):
+        """Return `True` if there are any missing class values."""
+        return self[self.domain.class_vars].isnull().any().any()
+
+    def iterrows(self):
+        """An override to return TableSeries instead of Series (pandas doesn't do that by default)."""
```

Comment:
This should be fixed by next pandas' release, probably. https://github.com/pydata/pandas/pull/13978